### PR TITLE
Add missing docstrings and typing

### PR DESCRIPTION
**Pull request by arturo-seijas**

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -118,29 +134,51 @@ def _are_pebble_instances_ready(self):
...
+        Returns:
+            The port number.
+        """
         site_url = self.config["site_url"]
         return urlparse(site_url).port
 
-    def _are_relations_ready(self, _):
-        """Handle the on pebble ready event for Indico."""
+    def _are_relations_ready(self, _) -> bool:
+        """Checks if the needed relations are stablished.
```
Comment (gregory-schiano) [inline src/charm.py:176]: Shoud it be `established` ?

File: src/charm.py [hunk 2]
```diff
@@ -435,8 +504,12 @@ def _get_indico_env_config(self, container):
             env_config = {**env_config, **self._get_http_proxy_configuration()}
         return env_config
 
-    def _get_http_proxy_configuration(self):
-        """Generate http proxy config."""
+    def _get_http_proxy_configuration(self) -> Dict:
+        """Generate http proxy config.
+
+        Returns:
+            Map containing the HTTP_PROXY environmental variables.
```
Comment (gregory-schiano) [inline src/charm.py:511]: I'd rather say `environment variables`

**Review by amandahla — DISMISSED** [review]
LGTM

File: src/charm.py (hunk 1 above)
Comment (arturo-seijas) [reply src/charm.py:176]: Good catch

**Review by gregory-schiano — APPROVED** [review]
LGTM

**Review by amandahla — APPROVED** [review]

**Comment by jdkandersson** [comment]
I realised this is merged, it might be worth adding `pydocstyle` which checks for things like missing an argument from the `Args` list etc

**Comment by jdkandersson** [comment]
> I realised this is merged, it might be worth adding `pydocstyle` which checks for things like missing an argument from the `Args` list etc

Submitted PR for this: https://github.com/canonical/indico-operator/pull/193

**Pull request merged**